### PR TITLE
Fix vctrs issues

### LINKED ISSUE #52: possible break in upcoming REDCapR v1.2.0
@RaymondBalise & others, a [recent change in REDCap v13.4.10](https://redcap.vanderbilt.edu/community/post.php?id=203708) required a change in REDCapR (see https://github.com/OuhscBbmc/REDCapR/issues/479).  

I ran `revdepcheck::revdep_details()` and saw this error for your package.  I'd like to release to CRAN this weekend.

Is this failed check related to (a) REDCap's changes, (b) REDCapR's changes, or (c) something else?

https://github.com/OuhscBbmc/REDCapR/commit/c601dcd99e69dcb3a84e57f7b1403e3e6bdee72d

<details>
* Version: 1.1.0
* GitHub: https://github.com/RaymondBalise/tidyREDCap
* Source code: https://github.com/cran/tidyREDCap
* Date/Publication: 2023-02-18 18:10:02 UTC
* Number of recursive dependencies: 82
Run `revdepcheck::revdep_details(, "tidyREDCap")` for more info
</details>

## Newly broken

*   checking tests ...
    ```
      Running 'testthat.R'
     ERROR
    Running the tests in 'tests/testthat.R' failed.
    Last 13 lines of output:
        8.       │ └─rlang::try_fetch(...)
        9.       │   └─base::withCallingHandlers(...)
       10.       └─tidyselect:::vars_select_eval(...)
       11.         └─tidyselect:::ensure_named(...)
       12.           └─vctrs::vec_as_names(names(pos), repair = "check_unique", call = call)
       13.             └─vctrs (local) `<fn>`()
       14.               └─vctrs:::validate_unique(names = names, arg = arg, call = call)
       15.                 └─vctrs:::stop_names_must_be_unique(names, arg, call = call)
       16.                   └─vctrs:::stop_names(...)
       17.                     └─vctrs:::stop_vctrs(...)
       18.                       └─rlang::abort(message, class = c(class, "vctrs_error"), ..., call = call)
      
      [ FAIL 1 | WARN 0 | SKIP 0 | PASS 4 ]
      Error: Test failures
      Execution halted
    ```

### PR DESCRIPTION
**Pull request by RaymondBalise**
This adds S3 methods that dplyr (and friends) need for working with labelled objects.  Closes #52

### CODE REVIEW

**Comment by RaymondBalise** [comment]
For more info on this look here:  https://github.com/r-lib/vctrs/issues/1833

File: R/labels.R [hunk 1]
```diff
@@ -1,44 +1,71 @@
...
 vec_ptype2.logical.labelled <- function(x, y, ...) {
   y
 }
 
+#### Casting to fix attributes ####
+# These functions all return the object `x` itself if it was already labelled 
+#   or `x` with the label from the `to` object.
 
-# both labelled
+##### both labelled #####
```
Comment (kyleGrealis) [inline R/labels.R:64]: these stand out much better now with the quad hashes

**Review by kyleGrealis — APPROVED** [review]
good notations and use of ####

File: R/labels.R (hunk 1 above)
Comment (gabrielodom) [reply R/labels.R:64]: This information should be added to the roxygen documentation using either the `@details` tag

**Review by gabrielodom — APPROVED** [review]
The only complaint I have is about the documentation, so this can be approved. 
Please move all the comments about how the `vctrs` extensions work to a `@details` section in the `R/labels.R` source.

**Pull request merged**